GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -714,27 +710,21 @@ private void SetShortcutKeyDisplayStringsFromHotkeySettings()
 
         private void RefreshWorkingDirCombo()
         {
-            Repository r = null;
-            if (Repositories.RepositoryHistory.Repositories.Count > 0)
-            {
-                r = Repositories.RepositoryHistory.Repositories[0];
-            }
+            var path = Module.WorkingDir;
 
-            List<RecentRepoInfo> mostRecentRepos = new List<RecentRepoInfo>();
-
-            if (r == null || !r.Path.Equals(Module.WorkingDir, StringComparison.InvariantCultureIgnoreCase))
-            {
-                Repositories.AddMostRecentRepository(Module.WorkingDir);
-            }
+            // save the current repository as the most recent
+            // the call is blocking because we are reloading the list of repositories
+            _repositoryHistory = ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.AddAsMostRecentAsync(path));
```

Comment:
I would not introduce `_repositoryHistory `. LoadHistoryAsync execution time is so short that it can be loaded each time it is needed. As a bonus we get always a fresh instance. Here we do not really need the whole list, it can be refactored later to not load it here at all.

Reply:
done